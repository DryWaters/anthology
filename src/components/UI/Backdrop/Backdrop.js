import React from "react";

import classes from "./Backdrop.module.scss";

const Backdrop = props =>
  props.show && <div className={classes.Backdrop} onClick={props.clicked} />;

export default Backdrop;
