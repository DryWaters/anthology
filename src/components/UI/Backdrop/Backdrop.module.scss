@import "../../../variables.scss";

.Backdrop {
  background-color: rgba($color-black, 0.7);
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 100;
}
