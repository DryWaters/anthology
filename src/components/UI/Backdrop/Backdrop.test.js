import React from "react";
import { shallow } from "enzyme";

import Backdrop from "./Backdrop";

describe("<Backdrop />", () => {
  it("should render if props.show is passed in", () => {
    const wrapper = shallow(<Backdrop show clicked={() => {}} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div")).toHaveLength(1);
  });

  it("should not render if props.show is false", () => {
    const wrapper = shallow(<Backdrop show={false} clicked={() => {}} />);
    expect(wrapper.find("div")).toHaveLength(0);
  });
});
