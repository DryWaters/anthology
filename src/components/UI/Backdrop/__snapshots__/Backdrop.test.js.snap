// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> should render if props.show is passed in 1`] = `ShallowWrapper {}`;
